### DISCUSSION TITLE
Disable virtual_text diagnostics for lua lsp

### DISCUSSION THREAD

**Discussion by gmnz** [NvChad/NvChad#1722] [Q&A] [answered]
Hey all,

I'm setting virtual_text = false in my lspconfig.lua like this:

```

local servers = {
  "html",
  "cssls",
  "volar",
  "intelephense",
  "tsserver",
  "tailwindcss",
}


for _, lsp in ipairs(servers) do
  lspconfig[lsp].setup {
    on_attach = on_attach,
    capabilities = capabilities,
    handlers = {
          ["textDocument/publishDiagnostics"] = vim.lsp.with(
             vim.lsp.diagnostic.on_publish_diagnostics, {
              -- Disable virtual_text
              virtual_text = false
             }
          ),
      }
  }
end

```

I'd like to do the same for the lua lsp but don't know how because it's "integrated" in the config... Could somebody please help me out? :-)

**Comment by siduck** [comment, marked as answer]
```lua
vim.diagnostic.config {      virtual_text = false      }
```

at the end of your custom lspconfig

**Reply by gmnz** [reply]
Great, thanks. :-) Really appreciate your "framework". Migrating from coc.nvim and this helps a lot. :-)

**Reply by siduck** [reply]
@gmnz btw that snippet worked because we set diagnostic config in our default lspconfig and when you add the snippet in your custom lspconfig, it overrides the default one
